src/mme-app/actionHandlers/attachActionHandlers.cpp

```diff
@@ -997,10 +1006,49 @@ ActStatus ActionHandlers::process_attach_cmp_from_ue(SM::ControlBlock& cb)
 ActStatus ActionHandlers::process_mb_resp(SM::ControlBlock& cb)
 {	
 	log_msg(LOG_DEBUG, "Inside handle_mb_resp \n");
+	
 	ProcedureStats::num_of_processed_mb_resp ++;
 	return ActStatus::PROCEED;
 }
 
+ActStatus ActionHandlers::send_emm_info(SM::ControlBlock& cb)
+{
+    log_msg(LOG_DEBUG, "Inside send_emm_info \n");
+
+    UEContext *ue_ctxt = static_cast<UEContext*>(cb.getPermDataBlock());
+    if (ue_ctxt == NULL)
+    {
+        log_msg(LOG_DEBUG, "send_emm_info: ue context is NULL \n");
+        return ActStatus::HALT;
+    }
+
+    struct ue_emm_info temp;
+    temp.msg_type = emm_info_request;
+    temp.enb_fd = ue_ctxt->getEnbFd();
+    temp.enb_s1ap_ue_id = ue_ctxt->getS1apEnbUeId();
+    temp.mme_s1ap_ue_id = ue_ctxt->getContextID();
+    temp.dl_seq_no = ue_ctxt->getDwnLnkSeqNo();
+    /*Logically MME should have TAC database. and based on TAC
+     * MME can send different name. For now we are sending Aether for
+     * all TACs
+     */
+    strcpy(temp.short_network_name, "Aether");
+    strcpy(temp.full_network_name, "Aether");
+    memcpy(&(temp.int_key), &((ue_ctxt->getUeSecInfo().secinfo_m).int_key),
+    NAS_INT_KEY_SIZE);
+
+    cmn::ipc::IpcAddress destAddr;
```

Comment:
Please use constructor to pass the arguments...

Reply:
Will take care of this as part of EDGEPOD-282